### PR TITLE
Profile360: Limit messaging to just one success, one error banner

### PR DESCRIPTION
**Pull request by ncksllvn**
Also modifies the `/v0/user` route to only add the cache-busting parameter when explicitly told. This way we can preserve caching except when we modify the data in Vet360.

Resolves department-of-veterans-affairs/vets.gov-team/issues/11297

### CODE REVIEW

File: src/applications/personalization/profile360/actions/updaters.js [hunk 1]
```diff
@@ -62,7 +62,8 @@ export function refreshTransaction(transaction, analyticsSectionName) {
       });
 
       if (isSuccessfulTransaction(transactionRefreshed)) {
-        dispatch(refreshProfile());
+        const forceCacheClear = true;
+        dispatch(refreshProfile(forceCacheClear));
       } else if (isFailedTransaction(transactionRefreshed) && analyticsSectionName) {
```
Comment (ncksllvn) [inline src/applications/personalization/profile360/actions/updaters.js]: Here's where we set that flag for clearing the cache.

File: src/applications/personalization/profile360/containers/Vet360TransactionReporter.jsx [hunk 2]
```diff
@@ -26,39 +28,43 @@ class Vet360TransactionReporter extends React.Component {
...
       </div>
     );
   }
 }
 
 const mapStateToProps = (state) => {
   return {
+    mostRecentSuccessfulTransaction: selectMostRecentSuccessfulTransaction(state),
+    mostRecentErroredTransaction: selectMostRecentErroredTransaction(state),
     successfulTransactions: selectVet360SuccessfulTransactions(state),
```
Comment (ncksllvn) [inline src/applications/personalization/profile360/containers/Vet360TransactionReporter.jsx]: Added new selectors for grabbing just the most recent successful/errored transaction. However, we still need to get the corresponding arrays, bc we need to clear all of them once the user clicks the close button.

File: src/applications/personalization/profile360/reducers/vet360.js [hunk 3]
```diff
@@ -10,10 +10,19 @@ import {
...
+
 const initialState = {
   transactions: [],
   fieldTransactionMap: {},
-  transactionsAwaitingUpdate: []
+  transactionsAwaitingUpdate: [],
+  metadata: {
+    mostRecentSuccessfulTransactionId: '',
+    mostRecentErroredTransactionId: '',
+  }
```
Comment (ncksllvn) [inline src/applications/personalization/profile360/reducers/vet360.js]: I'm not sure if `metadata` is the best term here, but I was trying to use a term broad enough that we could store other information as we need it.

File: src/applications/personalization/profile360/selectors.js [hunk 4]
```diff
@@ -49,6 +49,40 @@ export function selectVet360FailedTransactions(state) {
...
+    }
+  } = state;
+
+  let transaction = null;
+  if (mostRecentErroredTransactionId) {
+    transaction = transactions.find(t => t.data.attributes.transactionId === mostRecentErroredTransactionId);
+  }
+  return transaction;
+}
+
```
Comment (ncksllvn) [inline src/applications/personalization/profile360/selectors.js]: These are so verbose, but they get the point across.

**Review by bshyong — APPROVED** [review]
LGTM

**Pull request merged**